### PR TITLE
ui: make links clickable in public notes

### PR DESCRIPTION
**Pull request by oguzdemirbasci**
Ref: #1838

### CODE REVIEW

**Comment by drjova** [comment]
@oguzdemirbasci could you please add some tests?

File: ui/src/common/components/PublicNotesList.jsx [hunk 1]
```diff
@@ -4,19 +4,39 @@ import { List } from 'immutable';
...
 
+  static renderString(note) {
+    const parts = PublicNotesList.getValue(note).split(LINK_REGEX);
+    for (let i = 0; i < parts.length; i += 1) {
+      if (LINK_REGEX.test(parts[i])) {
+        parts[i] = (
+          <a
+            key={`link ${i}`}
+            href={parts[i]}
+            style={{ 'overflow-wrap': 'anywhere' }}
```
Comment (drjova) [inline ui/src/common/components/PublicNotesList.jsx:22]: we avoid using hardcoded style, we should move it to a separate file as a class

File: ui/src/common/components/PublicNotesList.jsx [hunk 2]
```diff
@@ -4,19 +4,39 @@ import { List } from 'immutable';
...
     return note.get('value');
   }
 
+  static renderString(note) {
+    const parts = PublicNotesList.getValue(note).split(LINK_REGEX);
+    for (let i = 0; i < parts.length; i += 1) {
+      if (LINK_REGEX.test(parts[i])) {
+        parts[i] = (
+          <a
+            key={`link ${i}`}
```
Comment (drjova) [inline ui/src/common/components/PublicNotesList.jsx:20]:
```suggestion
            key={`publication-note-link-${i}`}
```

File: ui/src/common/components/PublicNotesList.jsx [hunk 3]
```diff
@@ -4,19 +4,39 @@ import { List } from 'immutable';
...
+const LINK_REGEX = /(https?:\/\/(?:www\.|(?!www))[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\.[^\s]{2,}|www\.[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\.[^\s]{2,}|https?:\/\/(?:www\.|(?!www))[a-zA-Z0-9]+\.[^\s]{2,}|www\.[a-zA-Z0-9]+\.[^\s]{2,})/g;
+
 class PublicNotesList extends Component {
   static getValue(note) {
     return note.get('value');
   }
 
+  static renderString(note) {
+    const parts = PublicNotesList.getValue(note).split(LINK_REGEX);
+    for (let i = 0; i < parts.length; i += 1) {
```
Comment (drjova) [inline ui/src/common/components/PublicNotesList.jsx:16]: you could use `foreach`

File: ui/src/common/components/__tests__/__snapshots__/PublicNotesList.test.jsx.snap [hunk 4]
```diff
@@ -25,6 +25,25 @@ exports[`PublicNotesList renders public notes 1`] = `
       <span>
         note2
       </span>
+       • 
+    </li>
+    <li
+      key="note3 here https://indico.cern.ch/event/868940/contributions/3801009/attachments/2082517/3498118/Diversity_and_Inclusion_at_Belle_II_ICHEP_2020_Shanette_De_La_Motte.pdf"
+    >
```
Comment (drjova) [inline ui/src/common/components/__tests__/__snapshots__/PublicNotesList.test.jsx.snap:32]:
```suggestion
    this doesn't seem correct
```

File: ui/src/common/components/PublicNotesList.jsx (hunk 3 above)
Comment (oguzdemirbasci) [reply ui/src/common/components/PublicNotesList.jsx:16]: I need index too for the key.

File: ui/src/common/components/PublicNotesList/__tests__/__snapshots__/PublicNotesList.test.jsx.snap [hunk 5]
```diff
@@ -25,6 +25,21 @@ exports[`PublicNotesList renders public notes 1`] = `
       <span>
         note2
       </span>
+       • 
+    </li>
+    <li
+      key="note3 here https://pos.sissa.it/390/977/pdf"
```
Comment (drjova) [inline ui/src/common/components/PublicNotesList/__tests__/__snapshots__/PublicNotesList.test.jsx.snap:31]: I don't understand why the key is the link?

**Review by drjova — APPROVED** [review]

**Pull request merged**